### PR TITLE
Updates SyntaxStringBuilder to return itself

### PR DESCRIPTION
**Pull request by cheeezburga**
### Description
Makes the SyntaxStringBuilder's append methods return itself.
Allows for stuff nicer feeling stuff like:
```java
String toString = new SyntaxStringBuilder(event, debug)
    .append(x)
    .append("and")
    .append(y)
    .toString()
```

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

**Review by Moderocky — APPROVED** [review]

**Comment by Efnilite** [comment]
You can already chain arguments like this
```java
builder.append(x, "and", y);
```
Which is more readable than chained append calls on a single line (`builder.append(x).append("and").append(y)`) and the example you provided. This is something me and sovde have already discussed :)

This also encourages ternaries inside the append methods, as this encourages putting everything into a single expression, instead of being spread out over multiple statements with explicit `if`s, the latter of which has better readability. Essentially, this would allow the SSB to be used in the same way as the massive string concatenations used currently, but without the `toString(event, debug)`.

This is against the point of the SSB, which makes the `toString` more readable by forcing you to be explicit, which does sacrifice brevity.

Because of the unexpected popularity of the SSB, we may want to consider adding support for easy one-liners to it, e.g.
```java
new SyntaxStringBuilder(event, debug).single(x, "and", y);
```
Or similar. Personally, I support this, but a better solution should probably be discussed.

**Comment by cheeezburga** [comment]
> You can already chain arguments like this
> 
> ```java
> builder.append(x, "and", y);
> ```
> 
> Which is more readable than chained append calls on a single line (`builder.append(x).append("and").append(y)`) and the example you provided. This is something me and sovde have already discussed :)
I still think there's merit in having it return itself. The behaviour you describe is still there obviously, but I think this just adds an alternative which could be useful. If I'm alone in this line of thinking, then I can close it, but it seems like there's no downside to it.

**Comment by Fusezion** [comment]
> I still think there's merit in having it return itself. The behaviour you describe is still there obviously, but I think this just adds an alternative which could be useful. If I'm alone in this line of thinking, then I can close it, but it seems like there's no downside to it.

I prefer having `.append()` return, imo it's nicer and easier to read at times than seeing an array of arguments

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/lang/SyntaxStringBuilder.java [hunk 1]
```diff
@@ -30,31 +30,33 @@ public SyntaxStringBuilder(@Nullable Event event, boolean debug) {
...
 		}
+		return this;
 	}
 
 	/**
-	 * Adds multiple objects to the string.
+	 * Adds multiple objects to the string and returns the builder.
 	 * Spaces are automatically added between the provided objects.
 	 * @param objects The objects to add.
 	 */
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/lang/SyntaxStringBuilder.java:54]:
```suggestion
	 * Spaces are automatically added between the provided objects.
	 *
	 * @param objects The objects to add.
	 * @returns The builder.
	 */
```

File: src/main/java/ch/njol/skript/lang/SyntaxStringBuilder.java [hunk 2]
```diff
@@ -30,31 +30,33 @@ public SyntaxStringBuilder(@Nullable Event event, boolean debug) {
...
 
 	/**
-	 * Adds an object to the string.
+	 * Adds an object to the string and returns the builder.
 	 * Spaces are automatically added between the provided objects.
 	 * If the object is a {@link Debuggable} it will be formatted using
 	 * {@link Debuggable#toString(Event, boolean)}.
 	 *
 	 * @param object The object to add.
 	 */
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/lang/SyntaxStringBuilder.java:39]:
```suggestion
	 * @param object The object to add.
	 * @returns The builder.
	 */
```

**Review by Efnilite — CHANGES_REQUESTED** [review]

**Review by Efnilite — APPROVED** [review]

**Pull request merged**